### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

Keep adding test and improving code. fixing bugs and improving code quality.

### CODE REVIEW

File: src/Admin/PageAdmin.php [hunk 1]
```diff
@@ -115,14 +109,11 @@ protected function alterNewInstance(object $object): void
...
 
-            try {
-                $parent = $this->pageManager->getPageByUrl($site, implode('/', $slugs));
-            } catch (PageNotFoundException $e) {
-                try {
-                    $parent = $this->pageManager->getPageByUrl($site, '/');
-                } catch (PageNotFoundException $e) {
-                    throw new InternalErrorException('Unable to find the root url, please create a route with url = /');
-                }
+            $parent = $this->pageManager->getPageByUrl($site, implode('/', $slugs)) ??
```
Comment (jordisala1991) [inline src/Admin/PageAdmin.php:112]: This wasn't working before, getPAgeByUrl does not throw exception, but returns null.

File: src/Controller/BlockAdminController.php [hunk 2]
```diff
@@ -102,27 +103,32 @@ public function createAction(Request $request): Response
...
 
         $parent = $this->admin->getObject($parentId);
 
         if (null === $parent) {
-            throw new PageNotFoundException(sprintf('Unable to find parent block with id %d', $parentId));
+            throw new BadRequestHttpException(sprintf('Unable to find parent block with id: "%s"', $parentId));
         }
 
+        $this->admin->checkAccess('switchParent', $block);
+
```
Comment (jordisala1991) [inline src/Controller/BlockAdminController.php:131]: I think it looks better now.

File: src/Controller/BlockAdminController.php [hunk 3]
```diff
@@ -131,31 +137,30 @@ public function switchParentAction(Request $request): Response
...
+        $existingObject = $this->assertObjectExists($request, true);
+        \assert(null !== $existingObject);
 
-        $blockId = $request->get('block_id');
+        $this->checkParentChildAssociation($request, $existingObject);
 
-        $block = $this->admin->getObject($blockId);
+        $this->admin->checkAccess('composePreview', $existingObject);
 
-        if (null === $block) {
```
Comment (jordisala1991) [inline src/Controller/BlockAdminController.php:144]: If we use normal id parameter, and we use `assertObjectExists` like in the actions on CRUDController, we can control by default when the provided id exists and when not.

File: src/Controller/BlockAdminController.php [hunk 4]
```diff
@@ -131,31 +137,30 @@ public function switchParentAction(Request $request): Response
...
+     * @throws BadRequestHttpException
      */
     public function composePreviewAction(Request $request): Response
     {
-        $this->admin->checkAccess('composePreview');
+        $existingObject = $this->assertObjectExists($request, true);
+        \assert(null !== $existingObject);
 
-        $blockId = $request->get('block_id');
+        $this->checkParentChildAssociation($request, $existingObject);
```
Comment (jordisala1991) [inline src/Controller/BlockAdminController.php:147]: We were not even checking for child association before.

File: src/Admin/PageAdmin.php [hunk 5]
```diff
@@ -62,13 +61,8 @@ public function __construct(
...
     {
-        $collection->add('compose', $this->getRouterIdParameter().'/compose', [
-            $this->getIdParameter() => null,
-        ]);
-        $collection->add('compose_container_show', 'compose/container/'.$this->getRouterIdParameter(), [
-            $this->getIdParameter() => null,
-        ]);
-
+        $collection->add('compose', $this->getRouterIdParameter().'/compose');
+        $collection->add('compose_container_show', 'compose/container/'.$this->getRouterIdParameter());
```
Comment (jordisala1991) [inline src/Admin/PageAdmin.php:65]: There is no need to pass null by default.

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**